python-lib/dku_io_utils/recipe_config_loading.py

```diff
@@ -0,0 +1,289 @@
+import dataiku
+from dataiku.customrecipe import (
+    get_recipe_config,
+    get_input_names_for_role,
+    get_output_names_for_role,
+)
+import re
+from gluonts_forecasts.model_handler import list_available_models
+from dku_io_utils.partitions_handling import get_partition_root
+from constants import FORECASTING_STYLE_PRESELECTED_MODELS
+from safe_logger import SafeLogger
+
+logger = SafeLogger("Forecast plugin")
+
+
+class PluginParamValidationError(ValueError):
+    """Custom exception raised when the the plugin parameters chosen by the user are invalid"""
+
+    pass
+
+
+def load_training_config(recipe_config):
+    """Utility function to load, resolve and validate all training recipe config into a clean `params` dictionary
+
+    Returns:
+        Dictionary of parameter names (key) and values
+    """
+    params = {}
+
+    input_dataset_name = get_input_names_for_role("input_dataset")[0]
+    params["training_dataset"] = dataiku.Dataset(input_dataset_name)
+    training_dataset_columns = [p["name"] for p in params["training_dataset"].read_schema()]
+    params["partition_root"] = get_partition_root(params["training_dataset"])
+    check_equal_partition_dependencies(params["partition_root"], params["training_dataset"])
+
+    model_folder_name = get_output_names_for_role("model_folder")[0]
+    params["model_folder"] = dataiku.Folder(model_folder_name)
+
+    evaluation_dataset_name = get_output_names_for_role("evaluation_dataset")[0]
+    params["evaluation_dataset"] = dataiku.Dataset(evaluation_dataset_name)
+
+    params["make_forecasts"] = False
+    evaluation_forecasts_dataset_names = get_output_names_for_role("evaluation_forecasts_dataset")
+    if len(evaluation_forecasts_dataset_names) > 0:
+        params["evaluation_forecasts_dataset"] = dataiku.Dataset(evaluation_forecasts_dataset_names[0])
+        params["make_forecasts"] = True
+
+    params["time_column_name"] = recipe_config.get("time_column")
+    if params["time_column_name"] not in training_dataset_columns:
+        raise PluginParamValidationError("Invalid time column selection")
+
+    params["target_columns_names"] = sanitize_column_list(recipe_config.get("target_columns"))
+    if len(params["target_columns_names"]) == 0 or not all(
+        column in training_dataset_columns for column in params["target_columns_names"]
+    ):
+        raise PluginParamValidationError("Invalid target column(s) selection")
+
+    long_format = recipe_config.get("additional_columns", False)
+    if long_format:
+        params["timeseries_identifiers_names"] = sanitize_column_list(recipe_config.get("timeseries_identifiers", []))
+        if not all(column in training_dataset_columns for column in params["timeseries_identifiers_names"]):
+            raise PluginParamValidationError("Invalid timeseries identifiers column(s) selection")
```

Comment:
Could we log the selection?
```suggestion
            raise PluginParamValidationError(f"Invalid timeseries identifiers column(s) selection: {params['timeseries_identifiers_names']}")
```